src/libcollections/btree/node.rs

```diff
@@ -1526,6 +1526,22 @@ macro_rules! node_slice_impl {
                 (self.keys.len(), false)
             }
 
+            /// Performs linear search in a slice, from the back. Returns a tuple of
+            /// (index, is_exact_match).
+            fn search_linear_rev<Q: ?Sized>(&self, key: &Q) -> (usize, bool)
+                    where K: Borrow<Q>, Q: Ord {
+                let mut i = self.keys.len();
+                for k in self.keys.iter().rev() {
```

Comment:
Hm the reason is I'm sceptical of enumerate.rev. it's implementation is a bit messy. Of course, this scepticism may not be so justified.